chapter_01/final_exercise_01.md

```diff
@@ -21,66 +21,129 @@
 ### Chapter 1: Introduction to Big Data and Hadoop
 
 1. **Q:** What is Apache Hadoop?
+1. **A:** Hadoop is a distributed file system that works based on a master-slave architecture which was built with the assumption that hardware failure is the norm.
+
 2. **Q:** Can you name a few vendor-specific distributions of Hadoop?
+2. **A:** Cloudera Distribution Platform, Amazon Elastic MapReduce, Microsoft azure HDInsight.
+
 3. **Q:** What are the three modes in which Hadoop can run?
+3. **A:** **Standalone (default)** - where it runs as a single java process, **Psuedo Distributed** - where there is a single node cluster (many processes on one machine) which simulates the real world but still runs on one machine which is used to test and debug, **distributed** - a multi node cluster which includes the Namenode, Secondary Namenode and the Datanodes.
+
 4. **Q:** What is the primary role of the `hadoop-env.sh` configuration file?
+4. **A:** the primary role is to setup hadoop process and other java-related processes like memory for the java process.
+
 5. **Q:** What purpose does the `core-site.xml` file serve?
+5. **A:** The purpose of the core site is to configure mostly IO related things and also the filesystem and network addresses.
 
 ### Chapter 2: Hadoop Distributed File System (HDFS)
```

Comment:
If you don't know where to answer, answer at the top:
- Hdfs architecture(there are 2)
- Hdfs compared to s3
- Hdfs use cases
- Explain the following terms: failover, fencing, zkfc, quorum